proto/legacy_amino/amino.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package legacy_amino;
+
+import "google/protobuf/descriptor.proto";
+
+// TODO(fdymylja): once we fully migrate to protov2 the go_package needs to be updated.
+// We need this right now because gogoproto codegen needs to import the extension.
+option go_package = "github.com/cosmos/cosmos-sdk/types/tx/legacy_amino";
+
+extend google.protobuf.MessageOptions {
+  // name is the string used when registering a concrete
+  // type into the Amino type registry, via the Amino codec's
+  // `RegisterConcrete()` method. This string MUST be at most 39
+  // characters long, or else the message will be rejected by the
+  // Ledger hardware device.
+  string name = 11110001;
+
+  // encoding describes the encoding format used by Amino
+  // for the message. The field type is chosen to be a string so that
+  // the value can either be:
+  // - a machine-readable string, such as "base64", "bech32" or "utf8",
+  // - or a human-readable string, for instance a short specification of how
+  // a big integer would be encoded using Amino.
+  //
+  // If left empty, then the Amino encoding is expected to be the same as the
+  // Protobuf one.
+  //
+  // Please note that `cosmos_proto.ScalarDescriptor` has an option with the
+  // same name `legacy_amino_encoding`. The cosmos_proto one however is used
+  // as a protobuf _field_ annotation, whereas this one operates on a protobuf
+  // _message_.
+  string encoding = 11110002;
+}
+
+extend google.protobuf.FieldOptions {
+  // field_name sets a different field name (i.e. key name) in
+  // the amino JSON object for the given field.
+  //
+  // Example:
+  //
+  // message Foo {
+  //   string bar = 1 [(legacy_amino.field_name) = "baz"];
+  // }
+  //
+  // Then the Amino encoding of Foo will be:
+  // `{"baz":"some value"}`
+  string field_name = 11110003;
+
```

Comment:
I think we should move the `string encoding = 11110002` into the `FieldOptions` object. This would be more useful from what I understand.

Reply:
The idea proposed in this PR is:
- for `FieldOptions`: use scalars, see [example](https://github.com/cosmos/cosmos-sdk/pull/13466)
- for `MessageOptions`: use `legacy_amino.encoding`

Reply:
I hear there's some push back on using scalars. I'm fine to not use scalars for amino encoding. In this case, we introduce a new field option for amino encoding. But I still stand by [my point](https://github.com/cosmos/cosmos-sdk/pull/13501#discussion_r1006108527) that we _also_ need one for message options. Maybe:
- `legacy_amino.encoding` for FieldOptions
- `legacy_amino.message_encoding` for MessageOptions

Does that look better @pyramation ?

Reply:
ok great! yea this is good

I like these!

`legacy_amino.encoding` for FieldOptions
`legacy_amino.message_encoding` for MessageOptions

Reply:
yea for scalars I think we can hold off until we can all meet, but definitely the two options above can be used immediately and very atomically which I think is great.